File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        // vector<int> record;
        
        unordered_set<int> sequence;

        for(int i=0;i<nums.size();i++){
           sequence.insert(nums[i]);
                        

        }
        int longest = 0;
        for(int i=0;i<nums.size();i++){

            if(sequence.count(nums[i]-1) == 0){
                int lenght = 0;
                int j = 1;
                while(sequence.count(nums[i]+lenght)){
                    lenght++;                   
                }
                longest = max(lenght,longest);
                
            }
            

        }
        return longest;
    }
};

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count = {}
        l = 0
        r = 0
        final = 0
        # windows = 0
        while(r<len(s)):
            # window = (r - l) + 1
            count[s[r]] = 1 + count.get(s[r], 0)
            if(((r-l+1)-max(count.values())) <= k):
                final = max((r-l+1), final)
            
            else:
                count[s[l]] -= 1
                l+=1
            
            r+=1
        

        return final
                